qiskit/chemistry/results/electronic_structure_result.py

```diff
@@ -0,0 +1,346 @@
+    def nuclear_repulsion_energy(self, value: float) -> None:
+        """ Sets nuclear repulsion energy """
+        self.data['nuclear_repulsion_energy'] = value
+
+    @property
+    def nuclear_dipole_moment(self) -> Optional[DipoleTuple]:
+        """ Returns nuclear dipole moment X,Y,Z components in A.U when available from driver """
+        return self.get('nuclear_dipole_moment')
+
+    @nuclear_dipole_moment.setter
+    def nuclear_dipole_moment(self, value: DipoleTuple) -> None:
+        """ Sets nuclear dipole moment in A.U """
+        self.data['nuclear_dipole_moment'] = value
+
+    # TODO we need to be able to extract the statevector or the optimal parameters that can
+    # construct the circuit of the GS from here (if the algorithm supports this)
+
+    @property
+    def energy(self) -> Optional[float]:
+        """ Returns ground state energy if nuclear_repulsion_energy is available from driver """
+        # TODO the fact that this property is computed on the fly breaks the `.combine()`
+        # functionality
+        nre = self.nuclear_repulsion_energy
+        return self.electronic_energy + nre if nre is not None else None
+
+    @property
+    def electronic_energy(self) -> float:
+        """ Returns electronic part of ground state energy """
+        # TODO the fact that this property is computed on the fly breaks the `.combine()`
+        # functionality
+        return (self.computed_electronic_energy
+                + self.ph_extracted_energy
+                + self.frozen_extracted_energy)
+
+    @property
+    def computed_electronic_energy(self) -> float:
+        """ Returns computed electronic part of ground state energy """
+        return self.get('computed_electronic_energy')
+
+    @computed_electronic_energy.setter
+    def computed_electronic_energy(self, value: float) -> None:
+        """ Sets computed electronic part of ground state energy """
+        self.data['computed_electronic_energy'] = value
+
+    @property
+    def ph_extracted_energy(self) -> float:
+        """ Returns particle hole extracted part of ground state energy """
+        return self.get('ph_extracted_energy')
+
+    @ph_extracted_energy.setter
+    def ph_extracted_energy(self, value: float) -> None:
+        """ Sets particle hole extracted part of ground state energy """
+        self.data['ph_extracted_energy'] = value
+
+    @property
+    def frozen_extracted_energy(self) -> float:
+        """ Returns frozen extracted part of ground state energy """
+        return self.get('frozen_extracted_energy')
+
+    @frozen_extracted_energy.setter
+    def frozen_extracted_energy(self, value: float) -> None:
+        """ Sets frozen extracted part of ground state energy """
+        self.data['frozen_extracted_energy'] = value
+
+    # Dipole moment results. Note dipole moments of tuples of X, Y and Z components. Chemistry
+    # drivers either support dipole integrals or not. Note that when using Z2 symmetries of
```

Comment:
This comment appears to have been truncated

Reply:
The comment is identical to what is in the `MolecularGroundStateResult`. See here: https://github.com/Qiskit/qiskit-aqua/blob/6ce3b0c9f1a2b99dce3ee5677ed95a55b1738b83/qiskit/chemistry/core/chemistry_operator.py#L199

Reply:
Ok, I guess somehow it got truncated in the past. I imagine it would have been mentioning that with Z2Symmtries that is the dipole moment did not commute with the symmetry, then it cannot be measured and since no value can be determined for it it will be None to indicate this - i.e. unknown value, not measured.